feat: Add Prometheus MCP for AWS Managed Prometheus

MohamedSherifAbdelsamiea opened the pull request:
<!-- markdownlint-disable MD041 MD043 -->
Fixes

## Summary
This PR adds a new MCP for AWS Managed Prometheus, enabling users to execute PromQL queries, list metrics, and retrieve server information with AWS SigV4 authentication support.

## Features
- Execute instant PromQL queries
- Execute range queries with time intervals
- List available metrics
- Get server information
- AWS SigV4 authentication

## Dependencies
- boto3 >= 1.38.7
- httpx >= 0.28.1
- mcp[cli] >= 1.7.0
- requests >= 2.32.3

## Testing
The MCP has been tested with AWS Managed Prometheus workspaces in the following regions:
- us-east-1

## Documentation
Full documentation is included in the README.md file.

### Changes

Adding a new Prometheus MCP that provides a command-line interface for querying AWS Managed Prometheus with AWS SigV4 authentication, supporting both instant and range queries, metric listing, and server information retrieval.

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] I have reviewed the [contributing guidelines](https://github.com/awslabs/mcp/blob/main/CONTRIBUTING.md)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [ ] Changes are documented

Is this a breaking change? (Y/N)

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the [project license](https://github.com/awslabs/mcp/blob/main/LICENSE).

niallthomson commented:
I'm thinking it would be nice if I didn't have to create an MCP server for every AMP workspace I want to use? It might be a better user experience if the user can specify the AMP server by name and the MCP can resolve that to an endpoint by itself.

alexa-perlov commented:
Please take a look at the design guidelines and make sure you're following the standard project structure: https://github.com/awslabs/mcp/blob/main/DESIGN_GUIDELINES.md 

I'd recommend following these steps to use the cookie cutter template: https://github.com/awslabs/mcp/blob/main/DEVELOPER_GUIDE.md#working-on-your-server

niallthomson commented:
> I'm thinking it would be nice if I didn't have to create an MCP server for every AMP workspace I want to use? It might be a better user experience if the user can specify the AMP server by name and the MCP can resolve that to an endpoint by itself.

Also, please make the region a tool parameter instead of a global configuration parameter. It should be possible for the single server to hit different regions.

MohamedSherifAbdelsamiea commented on src/prometheus-mcp-server/awslabs/prometheus_mcp_server/server.py:
```diff
@@ -0,0 +1,498 @@
+            return False
+        
+        # Verify URL points to AWS Prometheus
+        if not ('amazonaws.com' in parsed_url.netloc and 'aps-workspaces' in parsed_url.netloc):
```
fixed it

alexa-perlov commented on src/prometheus-mcp-server/NOTICE:
```diff
@@ -0,0 +1,2 @@
+awslabs.short-domain-mcp-server
```
```suggestion
awslabs.prometheus-mcp-server
```

MohamedSherifAbdelsamiea replied:
fixed

alexa-perlov commented on src/prometheus-mcp-server/awslabs/prometheus_mcp_server/server.py:
```diff
@@ -0,0 +1,560 @@
+    """
+    try:
+        logger.info(f'Executing instant query: {query}')
+        params = {'query': query}
```
can we do query validation to prevent any dangerous queries?

alexa-perlov commented on src/prometheus-mcp-server/awslabs/prometheus_mcp_server/server.py:
```diff
@@ -0,0 +1,560 @@
+
+    Args:
+        endpoint: The Prometheus API endpoint to call
+        params: Query parameters to include in the request
```
can you do validation on the parameters?

alexa-perlov commented:
Can you pls take a look at docs/index.md and docs/servers and add an entry for the Prometheus MCP server. And can you add an entry for .github/CODEOWNERS. Thanks!

MohamedSherifAbdelsamiea commented:
> Can you pls take a look at docs/index.md and docs/servers and add an entry for the Prometheus MCP server. And can you add an entry for .github/CODEOWNERS. Thanks!

Created a new documentation file:

Added docs/servers/prometheus-mcp-server.md that includes the README content from the Prometheus MCP server

Updated the main documentation index:

Added an entry for the AWS Managed Prometheus MCP Server in docs/index.md

Included key features and a link to the detailed documentation

Updated the CODEOWNERS file:

Added an entry for /src/prometheus-mcp-server with you (@MohamedSherifAbdelsamiea) as the owner

Included the standard @awslabs/mcp-admins group as additional reviewers

alexa-perlov commented:
not a blocker for this PR, but consider adding in follow up PR(s): 

- Timeouts for query execution 
- Rate limiting

krokoko reviewed (approved):
LGTM, thanks ! Great checks on input validation and sanitization
For follow up PRs, we could also check/sanitize env var values, add some rate limiting and checks for the config file loaded
